KOGITO-7064 - Review Kogito persistence instructions adding Java serialization (#4207)
kiegroup/kie-docs: doc-content/kogito-docs/src/main/asciidoc/release-notes/ReleaseNotesKogito.1.22.0.Final/ReleaseNotesKogito.1.22.0.Final.adoc

// IMPORTANT: For 1.10 and later, save each version release notes as its own module file in the release-notes folder that this `ReleaseNotesKogito<version>.adoc` file is in, and then include each version release notes file in the chap-kogito-release-notes.adoc after Additional resources of {PRODUCT} deployment on {OPENSHIFT} section, in the following format:

> //include::release-notes/ReleaseNotesKogito<version>.adoc[leveloffset=+1]

Reviewer: ```suggestion
//include::ReleaseNotesKogito.<version>/ReleaseNotesKogito.<version>.adoc[leveloffset=+1]
```

Author: fixed

[id="ref-kogito-rn-new-features-1.22_{context}"]
= New features in {PRODUCT} 1.22

[role="_abstract"]
The following sections describe some of the new features or enhancements in {PRODUCT} 1.22.

== {PRODUCT} runtimes

=== Breaking change on {PRODUCT} runtime persistence

> Starting from {PRODUCT} 1.22.0, runtime persistence will be done using Java objects Serialization instead of custom marshallers and protobuf definitions.

Reviewer 2: ```suggestion
Starting from {PRODUCT} 1.22.0, runtime persistence will be done using Java objects serialization instead of custom marshallers and protobuf definitions.
```

This change requires that all process variables using custom data types implement `java.io.Serializable` and breaks compatibility with data persisted using

> previous versions of {PRODUCT} runtime persistence. It is recommended that process instances that still in progress complete before upgrading.

Reviewer 2: ```suggestion
This change requires all the process variables using custom data types to implement `java.io.Serializable` and breaks compatibility with data persisted using previous versions of {PRODUCT} runtime persistence. It is recommended that process instances that are still in progress are completed before upgrading.
```

// IMPORTANT: For 1.10 and later, save each version release notes as its own module file in the release-notes folder that this `ReleaseNotesKogito<version>.adoc` file is in, and then include each version release notes file in the chap-kogito-release-notes.adoc after Additional resources of {PRODUCT} deployment on {OPENSHIFT} section, in the following format:
//include::release-notes/ReleaseNotesKogito<version>.adoc[leveloffset=+1]

[id="ref-kogito-rn-new-features-1.22_{context}"]
= New features in {PRODUCT} 1.22

[role="_abstract"]
The following sections describe some of the new features or enhancements in {PRODUCT} 1.22.

== {PRODUCT} runtimes

=== {PRODUCT} runtime persistence allows to use Java serialization instead protobuf definitions

Starting from {PRODUCT} 1.22.0, runtime persistence will allow using Java objects serialization instead of custom marshallers and protobuf definitions.

> To enable this feature, simply set `kogito.persistence.proto.marshaller=false` in your application.properties file.

Reviewer 2: ```suggestion
To enable this feature, set `kogito.persistence.proto.marshaller=false` in your application.properties file.
```

Author: done
